core/tee/tee_svc_cryp.c

```diff
@@ -3487,6 +3605,7 @@ TEE_Result syscall_cryp_derive_key(unsigned long state,
 	const struct tee_cryp_obj_type_props *type_props = NULL;
 	TEE_Attribute *params = NULL;
 	size_t alloc_size = 0;
+	struct bbuf_obj *bbufs = NULL;
```

Comment:
I don't think this is needed. It doesn't need to be passed to any functions so it's only local usage that can be replaced by a couple of local helper variables.

Reply:
Okay. I will try to get rid of `bbufs`.

Reply:
Removed `bbufs` and made `get_*_params()` functions to use local variables or existing parameters as bounce buffers. `free()`s are missing, but these bounce buffers will be freed anyway when returning from the syscall.